pkg/frontend/querymiddleware/codec.go

```diff
@@ -755,6 +817,90 @@ func (c prometheusCodec) DecodeResponse(ctx context.Context, r *http.Response, _
 	return resp, nil
 }
 
+func (c prometheusCodec) DecodeLabelsQueryResponse(ctx context.Context, r *http.Response, lr LabelsQueryRequest, logger log.Logger) (Response, error) {
```

Comment:
Should this be split in half, one half for labels and another for series responses?

Or is there a reason to keep them in one method like this?

(same applies for `EncodeLabelsQueryResponse`)

Reply:
I kept it as one because the LabelsQueryRequest is used for label names, label values and series because their request formats are all fairly similar and this function outputs a Response which works for all three, and I don't want to clutter this with more functions that do almost the same thing, and only differentiate between the labels and series in the formatter interface (since the response formats are different).